lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -0,0 +1,409 @@
+    }
+
+    /**
+     * Migrate all old cache entries to new schema. No rollback supported.
+     * @param storeAuthStateInCookie
+     */
+    private migrateCacheEntries(): void {
+        const idTokenKey = `${Constants.CACHE_PREFIX}.${PersistentCacheKeys.ID_TOKEN}`;
+        const clientInfoKey = `${Constants.CACHE_PREFIX}.${PersistentCacheKeys.CLIENT_INFO}`;
+        const errorKey = `${Constants.CACHE_PREFIX}.${PersistentCacheKeys.ERROR}`;
+        const errorDescKey = `${Constants.CACHE_PREFIX}.${PersistentCacheKeys.ERROR_DESC}`;
+
+        const idTokenValue = this.browserStorage.getWindowStorageItem(idTokenKey);
+        const clientInfoValue = this.browserStorage.getWindowStorageItem(clientInfoKey);
+        const errorValue = this.browserStorage.getWindowStorageItem(errorKey);
+        const errorDescValue = this.browserStorage.getWindowStorageItem(errorDescKey);
+
+        const values = [idTokenValue, clientInfoValue, errorValue, errorDescValue];
+        const keysToMigrate = [PersistentCacheKeys.ID_TOKEN, PersistentCacheKeys.CLIENT_INFO, PersistentCacheKeys.ERROR, PersistentCacheKeys.ERROR_DESC];
+
+        keysToMigrate.forEach((cacheKey, index) => this.migrateCacheEntry(cacheKey, values[index]));
+    }
+
+    /**
+     * Utility function to help with migration.
+     * @param newKey
+     * @param value
+     * @param storeAuthStateInCookie
+     */
+    private migrateCacheEntry(newKey: string, value: string): void {
+        if (value) {
+            this.setItem(this.generateCacheKey(newKey), value, CacheSchemaType.TEMPORARY);
+        }
+    }
+
+    /**
+     * Parses key as JSON object, JSON.parse() will throw an error.
+     * @param key
+     */
+    private validateObjectKey(key: string): void {
+        JSON.parse(key);
+    }
+
+    /**
+     * Sets the cache item with the key and value given.
+     * Stores in cookie if storeAuthStateInCookie is set to true.
+     * This can cause cookie overflow if used incorrectly.
+     * @param key
+     * @param value
+     */
+    setItem(key: string, value: string | object, type: string): void {
+        // save the cacheItem
+        switch (type) {
+            case CacheSchemaType.ACCOUNT:
+            case CacheSchemaType.CREDENTIAL:
+            case CacheSchemaType.APP_METADATA:
+                this.browserStorage.setWindowStorageItem(key, JSON.stringify(value));
+                break;
+            case CacheSchemaType.TEMPORARY: {
+                const stringVal = value as string;
+                this.browserStorage.setWindowStorageItem(key, stringVal);
+                if (this.cacheConfig.storeAuthStateInCookie) {
+                    this.setItemCookie(key, stringVal);
+                }
+                break;
+            }
+            case CacheSchemaType.THROTTLING:
+            case CacheSchemaType.TELEMETRY: {
+                this.browserStorage.setWindowStorageItem(key, JSON.stringify(value));
+                break;
```

Comment:
can these two cases be combined with the ACCOUNT, CREDENTIAL, and APP_METADATA cases? they have exactly the same logic, as far as i can see

Reply:
I think this was potentially like this before this PR, but nonetheless